spring-graphql/src/main/java/org/springframework/graphql/execution/InvalidSchemaResourceException.java

```diff
@@ -0,0 +1,44 @@
+package org.springframework.graphql.execution;
+
+import graphql.GraphQLException;
+import graphql.schema.idl.errors.SchemaProblem;
+
+
+/**
+ * Indicates that the Resource in {@link GraphQlSource.Builder} is invalid.
+ *
+ * @author GenKui Du
+ * @since 1.0.0
+ */
+public class InvalidSchemaResourceException extends GraphQLException {
+
+    private final String resourceDescription;
+
+    private final SchemaProblem schemaProblem;
```

Comment:
The SchemaProblem will change internal to publicApi, we are free to use it, details in [issues 2734](https://github.com/graphql-java/graphql-java/issues/2734);